app/src/main/java/edu/wpi/first/shuffleboard/app/components/DashboardTabPane.java

```diff
@@ -102,7 +136,96 @@ public DashboardTab(String title) {
       setGraphic(new TabHandle(this));
 
       setWidgetPane(new WidgetPane());
+
       this.contentProperty().bind(widgetPane);
+
+      autoPopulate.addListener(__ -> populate());
+      sourcePrefix.addListener(__ -> populate());
+      availableSourceIds.addListener((ListChangeListener<String>) c -> populate());
+
+      setContextMenu(new ContextMenu(FxUtils.menuItem("Preferences", e -> {
+        // Use a dummy property here to prevent a call to populate() on every keystroke in the editor (!)
+        StringProperty dummySourcePrefix
+            = new SimpleStringProperty(sourcePrefix.getBean(), sourcePrefix.getName(), sourcePrefix.getValue());
+        WidgetPropertySheet propertySheet = new WidgetPropertySheet(
+            Arrays.asList(
+                this.title,
+                this.autoPopulate,
+                dummySourcePrefix,
+                getWidgetPane().tileSizeProperty()
+            ));
+        propertySheet.getItems().addAll(
+            new WidgetPropertySheet.PropertyItem<>(getWidgetPane().hgapProperty(), "Horizontal spacing"),
+            new WidgetPropertySheet.PropertyItem<>(getWidgetPane().vgapProperty(), "Vertical spacing")
+        );
+        Dialog<ButtonType> dialog = new Dialog<>();
+        dialog.setResizable(true);
+        dialog.titleProperty().bind(EasyBind.map(this.title, t -> t + " Preferences"));
+        dialog.getDialogPane().setContent(propertySheet);
+        dialog.getDialogPane().getButtonTypes().addAll(ButtonType.CLOSE);
+        dialog.setOnCloseRequest(__ -> {
+          this.sourcePrefix.setValue(dummySourcePrefix.getValue());
+        });
+        dialog.showAndWait();
+      })));
+    }
+
+    private void populate() {
+      if (getWidgetPane().getScene() == null || getWidgetPane().getParent() == null) {
+        // Defer until the pane is visible and is laid out in the scene
+        deferPopulation = true;
+        Platform.runLater(this::populate);
```

Comment:
I'm not very happy with this. I was counting 3000+ deferrals per startup... not fun.